### PR TITLE
ocamlPackages.odds: init at 1.2

### PR DESCRIPTION
**Pull request by Denommus**
https://github.com/raphael-proust/odds

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/ocaml-modules/odds/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  version = "1.2";
+
+  duneVersion = "3";
+
+  minimalOcamlVersion = "5.0.0";
+
+  src = fetchFromGitHub {
+    owner = "raphael-proust";
+    repo = pname;
+    rev = version;
```
Comment (steeleduncan) [inline pkgs/development/ocaml-modules/odds/default.nix:19]: `tag = `

**Review by steeleduncan — CHANGES_REQUESTED** [review]

**Comment by Denommus** [comment]
@steeleduncan comments addressed

File: pkgs/development/ocaml-modules/odds/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  fetchFromGitHub,
+  buildDunePackage,
+  menhir,
+  cmdliner,
+}:
+buildDunePackage rec {
+  pname = "odds";
+  version = "1.2";
+
+  duneVersion = "3";
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/odds/default.nix:12]: This is the default, please remove

**Comment by Denommus** [comment]
@ulrikstrid done

**Review by ulrikstrid — APPROVED** [review]

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368180`

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368180`

**Review by thiagokokada — APPROVED** [review]

File: pkgs/development/ocaml-modules/odds/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  fetchFromGitHub,
+  buildDunePackage,
+  menhir,
+  cmdliner,
+}:
+buildDunePackage rec {
+  pname = "odds";
+  version = "1.2";
+
+  minimalOcamlVersion = "5.0.0";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/odds/default.nix:12]: This is wrongly spelled.

File: pkgs/development/ocaml-modules/odds/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "raphael-proust";
+    repo = pname;
+    tag = version;
+    hash = "sha256-tPDowkpsJQKCoeuXOb9zPORoudUvkRBZ3OzkH2QE2zg=";
+  };
+
+  propagatedBuildInputs = [
+    cmdliner
+  ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/odds/default.nix:23]: This must not be propagated.

**Review by vbgl — COMMENTED** [review]
Can someone please fix this package?

**Comment by vbgl** [comment]
Done in #371357.

**Pull request merged**